internal/units/units.go

```diff
@@ -0,0 +1,214 @@
+// Package units provides functionality for parsing and displaying multiples of
+// bytes.
+package units
+
+import (
+	"encoding"
+	"errors"
+	"fmt"
+	"strconv"
+)
+
+var (
+	// ErrInvalidSyntax is returned when a byte string cannot be parsed.
+	ErrInvalidSyntax = errors.New("invalid syntax")
+
+	// ErrOverflow is returned when a byte string is too large to be represented.
+	ErrOverflow = errors.New("byte size overflows int64")
+)
+
+type Bytes int64
+
+var (
+	_ encoding.TextUnmarshaler = (*Bytes)(nil)
+	_ encoding.TextMarshaler   = Bytes(0)
+)
+
+const (
+	Byte Bytes = 1
+
+	Kilobyte = 1000 * Byte
+	Megabyte = 1000 * Kilobyte
+	Gigabyte = 1000 * Megabyte
+	Terabyte = 1000 * Gigabyte
+	Petabyte = 1000 * Terabyte
+	Exabyte  = 1000 * Petabyte
+
+	Kibibyte = 1024 * Byte
+	Mebibyte = 1024 * Kibibyte
+	Gibibyte = 1024 * Mebibyte
+	Tebibyte = 1024 * Gibibyte
+	Pebibyte = 1024 * Tebibyte
+	Exbibyte = 1024 * Pebibyte
+)
+
+var unitMap = map[string]Bytes{
+	"":   Byte,
```

Comment:
Is this a carryover from the old library? Ideally people would specify the unit

Reply:
Yeah, it's from the old library. It also affects the syntax's behavior of coercing a number to a string when assigning to a string attribute; so that setting `max_bytes = 15` remains valid